### ISSUE TITLE
regex being reported as polynomial but testing shows linear

### ISSUE THREAD

**Issue by pavanrust** [makenowjust-labs/recheck#847]
# What is the problem?

regex `\bPREFIX\b(?=([ \t]*))\1[^,\s]+(?=((?:[ \t]*,[ \t]*[^,\s]+)*[\t ]*\bSUFFIX\b))\2 `is reported as polynomial. I tried with the attack string, it does come linear to input length , here around 7700 steps . regex saved here with attack string input here. Please see ...
(https://regex101.com/r/kh6Nhh/1)

# How to fix the problem?

not sure
# Related issues

**Comment by makenowjust** [comment]
@pavanrust 

The analysis seems **partially** correct to me. At first, your regex is **non-linear**. Please take a look at the graph. It shows increasing matching time faster than linear. The last matching takes 3.6 seconds against the 100KB of string. This is unthinkably slow because my laptop can process 1GB of string in one second if a regex is linear.

<img width="534" alt="screenshot" src="https://user-images.githubusercontent.com/6679325/229273600-53bc95ac-7738-440b-9ca8-eb2a348ab9a6.png">

<details>
  <summary>Experiments Script</summary>

```js
const re = /\bPREFIX\b(?=([ \t]*))\1[^,\s]+(?=((?:[ \t]*,[ \t]*[^,\s]+)*[\t ]*\bSUFFIX\b))\2/;

for (let n = 0; n < 100; n++) {
  const s = 'P' + 'REFIX\x00UI,P'.repeat(n * 100) + '\n';
  const start = Date.now();
  re.exec(s);
  const time = Date.now() - start;
  console.log(`${s.length}\t${time}`);
}
```

</details>

However, the analysis may report its polynomial degree was $O(n^3)$ or $O(n^4)$, but it is too large to the actual degree. In this case, the correct complexity is $O(n^2)$ or $O(n^c)$ with $1 < c < 2$.

A regex containing non-regular extensions (e.g. back-reference) cannot be analyzed correctly for now (and the possibility of this is not known, just my research topic), so currently, `recheck` may report overapproximated results. This is a problem in principle and cannot be corrected.

If you have any questions, feel free to ask me. Or, if you have no questions, please close this issue.

Thank you.

**Comment by sudarshan12s** [comment]
Thanks for details. Yes I see If i increase the length of attack string hinted by tool,  it seems to take longer time .  I shall close this issue. 

```
'I' + ')PREFIX'.repeat(1024) + 'R'
'I' + ')PREFIX'.repeat(10240) + 'R'
```

**Comment by pavanrust** [comment]
Thanks for sharing details. Yes when I increase the length of the attack String hinted, it does take longer time . got it now. closing issue. 

`'I' + ')PREFIX'.repeat(10240) + 'R';`
